test/test_type_promotion.py

```diff
@@ -79,15 +79,26 @@ def test_int_promotion(self, device):
         self.assertEqual(c.dtype, torch.int64)
 
     @float_double_default_dtype
-    def test_float_promotion(self, device):
-        a = torch.ones([4, 4, 4], dtype=torch.float, device=device)
-        b = torch.ones([4, 4, 4], dtype=torch.double, device=device)
-        c = a + b
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
-        c = b + a
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
+    def test_complex_float_promotion(self, device):
```

Comment:
if all of the basic math operations are now supported, I wonder if we should add complex dtypes to torch.testing.get_all_math_dtypes. This would then enable tests like TestTypePromotion.test_many_promotions (in this file) to run on complex types.

Reply:
prob no in this PR because, torch.testing.get_all_math_dtypes is being currently used in some test functions in test_nn.py and test_torch.py which are not defined for complex numbers as yet eg. test_cosine_embedding_loss_with_diff_type . perhaps we can add get_all_math_detypes_and_complex for now?